### PR TITLE
adding option '--insecure-policy' for passthrough and reencrypt routes

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
PR11953 added support for insecureEdgeTerminationPolicy to reencrypt and passthrough routes,
so it should be supported in the CLI as well.

Bug: 1403155 [Link](https://bugzilla.redhat.com/show_bug.cgi?id=1403155)

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@openshift/networking

**Comment by knobunc** [comment]
[test] @ramr PTAL

**Comment by JacobTanenbaum** [comment]
The BZ that ben mentioned above [BZ1418023](https://bugzilla.redhat.com/show_bug.cgi?id=1418023)

File: docs/man/man1/oc-create-route-passthrough.1 [hunk 1]
```diff
@@ -33,6 +33,10 @@ Specify the service (either just its name or using type/name syntax) that the ge
     Set a hostname for the new route
 
 .PP
+\fB\-\-insecure\-policy\fP=""
+    Set an insecure for the new route
```
Comment (ramr) [inline docs/man/man1/oc-create-route-passthrough.1]: Set an insecure `policy` (missing that word) for the new route.

File: pkg/cmd/cli/cmd/create/route.go [hunk 2]
```diff
@@ -198,6 +198,7 @@ func NewCmdCreatePassthroughRoute(fullName string, f *clientcmd.Factory, out io.
 	kcmdutil.AddDryRunFlag(cmd)
 	cmd.Flags().String("hostname", "", "Set a hostname for the new route")
 	cmd.Flags().String("port", "", "Name of the service port or number of the container port the route will route traffic to")
+	cmd.Flags().String("insecure-policy", "", "Set an insecure for the new route")
```
Comment (ramr) [inline pkg/cmd/cli/cmd/create/route.go]: Set an insecure `policy` (missing keyword!) for the new route.

**Comment by ramr** [comment]
@knobunc as I commented on the other PR - adding some of these checks on client side could be problematic. One is that older cli clients would behave differently vis-a-vis newer cli clients + what happens when the server validation changes say for a list of choices from [foo, bar] to [foo, bar, baz] or even [one, two]? The older cli client checks would be invalid at that point or worse yet even block valid values.

**Comment by knobunc** [comment]
[merge]

**Comment by JacobTanenbaum** [comment]
[test]

**Comment by JacobTanenbaum** [comment]
@openshift/cli-review PTAL

**Comment by knobunc** [comment]
[test]

**Comment by juanvallejo** [comment]
LGTM

**Comment by knobunc** [comment]
[merge]

**Comment by smarterclayton** [comment]
We need less client side validation, and more serverside.  Clients should not make decisions about valid values.

**Comment by knobunc** [comment]
Ok, based on @smarterclayton's feedback, I have closed the above mentioned bug since we are functioning as designed.

**Comment by knobunc** [comment]
[merge] Last flaked against #12736.  Logs: https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_integration/10912/consoleFull#-83175598956bf4006e4b05b79524e5923

**Pull request merged**